risc0/circuit/rv32im/src/prove/emu/exec/mod.rs

```diff
@@ -541,15 +540,23 @@ impl<'a, 'b, S: Syscall> Executor<'a, 'b, S> {
     fn ecall_bigint2(&mut self) -> Result<bool> {
         let blob_ptr = self.load_guest_addr_from_register(REG_A0)?.waddr();
         let nondet_program_ptr = self.load_guest_addr_from_register(REG_T1)?;
-        let nondet_program_size = self.load_u32_from_guest((blob_ptr + 0u32).baddr())?;
+        let verify_program_ptr = self.load_guest_addr_from_register(REG_T2)?;
```

Comment:
Maybe we should schedule a discussion around registers/calling convention for bigint2 sometime before we get to a release, just to make sure we are happy with things as it sort of becomes part of the ABI we might need to support in the future (although maybe it's not relevant since we can make a 'compatibility' kernel for v2, and change whatever we want).